frontend/src/pages/_app.tsx

```diff
@@ -4,25 +4,30 @@ import Head from 'next/head'
 import { NextComponentType, NextPageContext } from 'next/types'
 
 import { LayoutDefault } from '@/components'
+import { useAuth } from '@/hooks'
 
 type CustomAppProps = AppProps & {
   Component: NextComponentType<NextPageContext, any, any> & {
-    title?: string,
-    layout?: ({ children }: { children: React.ReactNode }) => JSX.Element,
+    title?: string
+    layout?: ({ children }: { children: React.ReactNode }) => JSX.Element
   }
 }
 
 export default function MyApp({ Component, pageProps }: CustomAppProps) {
-  const title = Component.title ? `Book Store | ${Component.title}` : 'Book Store'
-  
-  const Layout = Component.layout ? Component.layout  : LayoutDefault;
+  const { user, onLogout } = useAuth()
+
+  const title = Component.title
+    ? `Book Store | ${Component.title}`
+    : 'Book Store'
+
+  const Layout = Component.layout ? Component.layout : LayoutDefault
```

Comment:
I will make it cleaner in the future

Reply:
ok. thank you. I know this is just the simple example, but at the next time please upgrade better. 👍